src/serialization/KiotaSerializer.Deserialization.cs

```diff
@@ -31,7 +31,7 @@ private static Stream GetStreamFromString(string source)
     {
         var stream = new MemoryStream();
         using var writer = new StreamWriter(stream, Encoding.UTF8, 1024, true);
-        writer.WriteAsync(source).GetAwaiter().GetResult(); // so the asp.net projects don't get an error
```

Comment:
please revert as this new implementation will cause errors for clients used in an asp.net mvc application which doesn't allow reading network streams synchronously (default)

Reply:
From what I can tell, this isn't a network stream, it's all local to this serializer and it's based on the `source` string and based on a `MemoryStream`. What am I missing?

Reply:
maybe this is indeed only required when reading/writing from a network stream. I can't remember the specifics here.
@andrueatman, do you remember what the issue was?

Reply:
I can't remember clearly as well. 
I think maybe the right thing to do here is probably keep this as is for now and look into exposing async overloads for the versions of the API that take in a stream param?

Reply:
Yeah, I agree that we should expose async overloads. And we should avoid _sync over async_ at all costs, as it can be the cause of deadlocks and is an antipattern.